core/src/wallet/bitcoin/BitcoinLikeAccount.cpp

```diff
@@ -708,5 +707,195 @@ namespace ledger {
+                        }
+                    }
+                }
+
+                // Get outputs value and change output if it exists
+                BigInt outputsValues;
+                std::shared_ptr<api::BitcoinLikeOutput> outputChange;
+                for (auto &o : tx->getOutputs()) {
+                    // Update outputs value
+                    outputsValues = outputsValues + *(std::dynamic_pointer_cast<ledger::core::Amount>(o->getValue())->value());
+
+                    // Get change output
+                    auto isChange = self->getKeychain()->getAddressPurpose(o->getAddress().value_or(""))
+                                            .getValueOr(BitcoinLikeKeychain::KeyPurpose::RECEIVE) == BitcoinLikeKeychain::KeyPurpose::CHANGE;
+                    // We take last output to a change address
+                    outputChange = isChange ? o : outputChange;
+                }
+
+                // Get new fees per byte
+                auto fees = inputsValue - outputsValues;
+                auto additionalFees = static_cast<uint64_t>(additionaFeesPerBytes->toLong() * rawTx.size());
+
+                auto incrementSequence = [=](const std::shared_ptr<BitcoinLikeTransactionApi> &transaction) {
+                    for (auto &input : transaction->getInputs()) {
+                        // TODO: support uints in Djinni
+                        std::dynamic_pointer_cast<BitcoinLikeWritableInputApi>(input)->setSequence(input->getSequence() + 1);
+                    }
+                };
+
+                // First we try to get change output and decrease it's amount
+                // Case change output can cover new fees, in this case we decrease it's amount
+                if (outputChange) {
+                    if (outputChange->getValue()->toLong() > additionalFees) {
+                        BitcoinLikeBlockchainExplorerOutput out;
+                        out.address = outputChange->getAddress();
+                        out.index = outputChange->getOutputIndex();
+                        out.value = BigInt(outputChange->getValue()->toString()) - BigInt((unsigned long long) additionalFees);
+                        out.transactionHash = outputChange->getTransactionHash();
+                        out.script = hex::toString(outputChange->getScript());
+                        tx->addOutput(std::make_shared<BitcoinLikeOutputApi>(out, self->getWallet()->getCurrency()));
+                        tx->removeOutput(outputChange->getOutputIndex());
+                        incrementSequence(tx);
+                        return FuturePtr<api::BitcoinLikeTransaction>::successful(tx);
+                    }
+                }
+
+                // Construct new outputs taking into account fees
+                // Pick new UTXO
+                return self->getUTXO().mapPtr<api::BitcoinLikeTransaction>(self->getContext(), [=](const std::vector<std::shared_ptr<api::BitcoinLikeOutput>> &utxos) {
```

Comment:
If we have a change in the transaction that we want to modify, this change will be in the list of UTXOs (right?), you don't take this into account when aggregating UTXOs to adjust the sum, and later you try to change the amount of the change.

Reply:
So again since we are taking the deepest first we should be safe unless the only availale UTXO is the one generated by the previous tx we want tto use RBF on, in which case it will always fail ... 
But we can explicitly add a test and fail on creating RBF transaction in that case 🤔